tests/ui/privacy/unnameable_types.rs

```diff
@@ -0,0 +1,30 @@
+#![allow(unused)]
+#![allow(private_in_public)]
+#![deny(unnameable_types)]
+
+mod m {
+    pub struct PubStruct(pub i32); //~ ERROR struct `PubStruct` is reachable but cannot be named
+
+    pub enum PubE { //~ ERROR enum `PubE` is reachable but cannot be named
+        V(i32), //~ ERROR variant `PubE::V` is reachable but cannot be named
```

Comment:
Let's drop this diagnostic for enum variants after all.
If it's almost always duplicated with the same diagnostic for enums it doesn't look good.